runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/RunProcessor.java

```diff
@@ -0,0 +1,71 @@
+package uk.ac.bbsrc.tgac.miso.runscanner;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.TimeZone;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.prometheus.client.Counter;
+import io.prometheus.client.Histogram;
+import uk.ac.bbsrc.tgac.miso.core.data.type.PlatformType;
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
+import uk.ac.bbsrc.tgac.miso.dto.NotificationDto;
+
+public abstract class RunProcessor {
+  private static final Counter errors = Counter.build().name("miso_runscanner_errors").help("The number of bad directories encountered.")
+      .labelNames("platform").register();
+
+  public static final Iterable<RunProcessor> INSTANCES = Arrays.asList();
+
+  private static Logger log = LoggerFactory.getLogger(RunProcessor.class);
+
+  private static final Histogram processTime = Histogram.build().buckets(1, 5, 10, 30, 60, 300, 600, 3600)
+      .name("miso_runscanner_directory_process_time").help("Time to process a run directories in seconds.")
+      .labelNames("platform", "instrument").register();
+
+  private final String name;
+
+  private final PlatformType platformType;
+
+  public RunProcessor(PlatformType platformType, String name) {
+    super();
+    this.platformType = platformType;
+    this.name = name;
```

Comment:
It's probably going to be GenomeAnalyzer II versus every other instrument in the case of Illumina. For the PacBio, it's probably going to reflect the version of the webservice rather than the version of the instrument itself.